src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java

```diff
@@ -372,6 +376,8 @@ private AIState lookForNeededItems()
 
     /**
      * Updates the itemsCurrentlyNeeded with current values.
+     * 
+     * This function seems bugged, as it gives itemsCurrentlyNeeded = isEmpty 
```

Comment:
If the builder is able to execute one action then for sure the list should be cleared that's correct behavior. If he requires an item he won't work.
I think I'll get to test that today still. I'll take a look.
But can you please tell me a guaranteed reproduce of incorrect behavior?